src/helpers/stateful_list.rs

```diff
@@ -77,4 +78,26 @@ impl<T> StatefulList<T> {
     pub fn unselect(&mut self) {
         self.state.select(None);
     }
+    pub fn select(&mut self, i: usize) {
+        self.curr = i;
+        self.state.select(Some(i));
+    }
+}
+
+impl<T: ToString> StatefulList<T> {
+    pub fn find_by_path(&self, s: &PathBuf) -> usize {
+        let mut i = 0;
+
+        for n in 0 .. self.items.len() {
+            if s.ends_with(self.items[n].to_string()) {
```

Comment:
I didn't think this through. Not sure if this satisfies every use case, or even if it'd never throw an error or crash the app.

"It works in my machine" is all I can say :see_no_evil: